easybuild/tools/module_generator.py

```diff
@@ -226,18 +226,21 @@ def prepend_paths(self, key, paths, allow_abs=False):
             self.log.debug("Wrapping %s into a list before using it to prepend path %s" % (paths, key))
             paths = [paths]
 
-        for i, path in enumerate(paths):
```

Comment:
we were updating elements in a list while iterating over it, which is OK

What's not OK is that we're modifying the list that is being passed to this method...